srtcore/core.cpp

```diff
@@ -614,7 +614,8 @@ void srt::CUDT::getOpt(SRT_SOCKOPT optName, void *optval, int &optlen)
         }
 
         // Fallback: return from internal data
-        strcpy(((char*)optval), m_config.sBindToDevice.c_str());
+        strncpy(((char*)optval), m_config.sBindToDevice.c_str(), (size_t)optlen - 1);
+        ((char*)optval)[optlen - 1] = '\0';
```

Comment:
This should also work and look much better. I didn't check it though.
```suggestion
        m_config.sBindToDevice.copy((char*)optval, optlen - 1);
```

Reply:
sBindToDevice is a std::string so I guess it would be std::string::assign or something. Will have a look, as I agree it's way neater.

The other similar strncpy calls in that part of the code use StringStorage which can be changed to (for example(

```
m_config.sPacketFilterConfig.set((char*)optval, optlen - 1);
```

Reply:
According to this:

https://cplusplus.com/reference/string/string/copy/

Your suggestion would still need the explicit null terminator line, so I don't think it adds much.

Reply:
Yes, but at the right place, not extra.
```
  std::size_t length = str.copy(buffer,6,5);
  buffer[length]='\0';
```